sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala

```diff
@@ -40,6 +41,12 @@ import org.apache.spark.sql.types._
   group = "string_funcs")
 object CollateExpressionBuilder extends ExpressionBuilder {
   override def build(funcName: String, expressions: Seq[Expression]): Expression = {
+    // Needed in order to make sure no parse errors appear when Collation Support is not enabled
+    // because this might suggest to the customer that feature is enabled until they reproduce
+    // the right syntax
+    if (!SQLConf.get.collationEnabled) {
+      throw QueryCompilationErrors.collationNotEnabledError()
+    }
```

Comment:
When is this executed? How often?
How come we do not block this at the analyzer level?

Reply:
Changed now, but needed to keep some check in the builder, as analyzer did not fail if collation name was wrong, and we would give confusing messages to the user in that case. I presume we do not want to give them suggestions on how to fix syntax if collation is disabled.